src/sourmash/sig/grep.py

```diff
@@ -97,9 +97,8 @@ def main(args):
         elif not args.silent:
             # nope - do output signatures. convert manifest to picklist, apply.
             sub_picklist = sub_manifest.to_picklist()
-
             try:
-                idx = idx.select(picklist=sub_picklist)
+                idx = idx.select(picklist=sub_picklist, filter_manifest=sub_manifest)
```

Comment:
@ctb I thought I might be able to get rid of using the `picklist=sub_picklist` here, but I think it's needed for LCA databases and databases with no manifests.